setup.py

```diff
@@ -53,12 +53,12 @@
 """
 
 install_requires = [
-    "trio >= 0.12.0",
-    "async_generator >= 1.6",
+    "trio >= 0.15.0",
```

Comment:
why is it necessary to require trio 0.15?

generally python libraries should be as flexible as possible about required dependencies, otherwise it causes headaches for developers

here it's certainly possible to import form `lowlevel` and fallback to `hazmat`

e.g. see https://github.com/groove-x/trio-util/pull/5

Reply:
> generally python libraries should be as flexible as possible about required dependencies, otherwise it causes headaches for developers

For example, if a large trio codebase is wanting to upgrade trio-asyncio for a feature or bugfix, they are compelled to upgrade trio at the same time.

Reply:
@belm0 What Trio/Python version are you using and what is holding you back?

I know @njsmith is prioritizing changes that break backwards-compatibility, but until we hit 1.0 we'll continue to break things like this, and maintaining support for older versions of Trio also has a cost, like multiplying the number of jobs we have to run in CI for every Trio project.

Reply:
I'm mentioning this as a library author (e.g. trio-util):  **it's a courtesy to your users (developers) to have the library support some modest amount of legacy**.  Trio itself practices this in the form of supporting legacy Python versions.  I'm not suggesting trio-asyncio support every past version of Trio, but supporting at least one breaking version back would be a good value.

Why:  because an application might use *multiple* libraries which depend on Trio, and those libraries may vary widely in how fast their maintainers incorporate the latest Trio and publish.  Following your approach of "only support the latest Trio", such apps would be at the whim of the laggard library and couldn't update any of the libraries depending on Trio until all of them supported the new version.

I understand Trio employs deprecation, but the warning messages are annoying to development and can cause developers to overlook actionable warnings which aren't caused by transitive dependencies.

Reply:
@belm0 Thanks for explaining, that makes sense. Now that trio 0.16.0 is out, would you consider supporting 0.15.0+ acceptable, or should we only require 0.15.0+ in a few weeks?

Reply:
I don't know if a rule like "one version back" makes sense, given that changes and release frequency of Trio are arbitrary.

What matters is the Trio breaking changes affecting trio-asyncio.  I'm guessing with the `lowlevel` fallback to hazmat, you could probably support back to Trio 0.11 or something.

What I did with my libraries:  when adding support for a breaking change like hazmat rename, make a copy of test-requirements.txt and name it according to the Trio version it was using, e.g. test-requirements_trio-0.11.txt.  The have CI run against the old and new test requirements.

Reply:
Supporting a larger range of versions is definitely a nice thing, but it also has a significant cost (extra testing, making and removing shims, etc.). I think in this particular case, we shouldn't spend energy worrying about it, because trio-asyncio is struggling to support one version of Trio. And Trio itself already does a lot of work to support laggard libraries – that's why we work hard to make deprecation periods :-). Putting the back-compat shims in Trio is more efficient, since we only have to do it once, rather than in every downstream library.

Deprecations do issue warnings, but if that's a problem then you can use the `warnings` module's filtering options to hide `TrioDeprecationWarning`, in general, or in specific deployments, or only see the ones that are triggered by your module and ignore the rest.

Reply:
(Also of course we'll want to re-evaluate this over time; when the ecosystem gets more mature then the balance shifts more towards conservative changes and supporting broader version windows.)